### PR TITLE
CTCTRADERS-1097: Add GoodsItem model

### PR DESCRIPTION
**Pull request by chris-ianson**
All models added that build up a GoodsItem record.

Note: SpecialMention and Package have json reads/writes and may need updating depending on how we store the values for these objects within the db (won't know until the goods items journey is started)

### CODE REVIEW

File: app/models/messages/goodsitem/Package.scala [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+    import com.lucidchart.open.xtract.__
+
+    (
+      (__ \ "KinOfPacGS23").read[String],
+      (__ \ "MarNumOfPacGS21").read[String].optional,
+    ).mapN(apply)
+  }
+
+  val validCodes: Set[String] = Set("VQ", "VG", "VL", "VY", "VR", "VS", "VO")
```
Comment (MatthewAlexanderBeck) [inline app/models/messages/goodsitem/Package.scala:79]: Is this something we might be interested putting into AppConfig or are they unlikely to change?

File: app/models/messages/goodsitem/Package.scala [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    import com.lucidchart.open.xtract.__
+
+    (
+      (__ \ "KinOfPacGS23").read[String],
+      (__ \ "NumOfPieGS25").read[Int],
+      (__ \ "MarNumOfPacGS21").read[String].optional,
+    ).mapN(apply)
+  }
+
+  val validCodes: Set[String] = Set("NE", "NF", "NG")
```
Comment (MatthewAlexanderBeck) [inline app/models/messages/goodsitem/Package.scala:136]: Same here?

File: app/models/messages/goodsitem/TraderConsigneeGoodsItem.scala [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    (__ \ "TINCE259").read[String].optional
+  ).mapN(apply)
+
+  implicit def writes: XMLWrites[TraderConsigneeGoodsItem] = XMLWrites[TraderConsigneeGoodsItem] {
+    trader =>
+      <TRACONCE2>
+        <NamCE27>{escapeXml(trader.name)}</NamCE27>
+        <StrAndNumCE222>{trader.streetAndNumber}</StrAndNumCE222>
+        <PosCodCE223>{trader.postCode}</PosCodCE223>
+        <CitCE224>{trader.city}</CitCE224>
```
Comment (MatthewAlexanderBeck) [inline app/models/messages/goodsitem/TraderConsigneeGoodsItem.scala:52]: Do any of these need to be escapeXml too?

File: app/models/messages/goodsitem/TraderConsignorGoodsItem.scala [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  ).mapN(apply)
+
+  implicit def writes: XMLWrites[TraderConsignorGoodsItem] = XMLWrites[TraderConsignorGoodsItem] {
+    trader =>
+      <TRACONCO2>
+        <NamCO27>{escapeXml(trader.name)}</NamCO27>
+        <StrAndNumCO222>{trader.streetAndNumber}</StrAndNumCO222>
+        <PosCodCO223>{trader.postCode}</PosCodCO223>
+        <CitCO224>{trader.city}</CitCO224>
+        <CouCO225>{trader.countryCode}</CouCO225>
```
Comment (MatthewAlexanderBeck) [inline app/models/messages/goodsitem/TraderConsignorGoodsItem.scala:53]: Same here

**Review by MatthewAlexanderBeck — APPROVED** [review]

File: app/models/messages/goodsitem/Package.scala (hunk 1 above)
Comment (chris-ianson) [reply app/models/messages/goodsitem/Package.scala:79]: This is set in the DDNTA which hasn't changed much (on the surface, anyway!) I don't think these will change so happy to have the values hardcoded (if we find out during testing or in live that the spec is wrong and they need changing we can consider putting them into a conf file)

File: app/models/messages/goodsitem/TraderConsigneeGoodsItem.scala (hunk 3 above)
Comment (chris-ianson) [reply app/models/messages/goodsitem/TraderConsigneeGoodsItem.scala]: I've added escapeXml around street and city for all trader models

**Pull request merged**